crates/core/tedge/src/cli/certificate/create.rs

```diff
@@ -62,20 +79,42 @@ impl CreateCertCmd {
         validate_parent_dir_exists(&self.cert_path).map_err(CertError::CertPathError)?;
         validate_parent_dir_exists(&self.key_path).map_err(CertError::KeyPathError)?;
 
-        let cert = KeyCertPair::new_selfsigned_certificate(config, &self.id, key_kind)?;
-
-        // TODO cope with broker user being tedge
-        // Creating files with permission 644 owned by the MQTT broker
-        let mut cert_file =
-            create_new_file(&self.cert_path, crate::BROKER_USER, crate::BROKER_GROUP)
-                .map_err(|err| err.cert_context(self.cert_path.clone()))?;
-
-        let cert_pem = cert.certificate_pem_string()?;
-        cert_file.write_all(cert_pem.as_bytes())?;
-        cert_file.sync_all()?;
-
-        // Prevent the certificate to be overwritten
-        set_permission(&cert_file, 0o444)?;
+        let cert = match &self.csr_path {
+            Some(csr_path) => {
+                validate_parent_dir_exists(csr_path).map_err(CertError::CsrPathError)?;
+
+                let cert = KeyCertPair::new_certificate_sign_request(config, &self.id, key_kind)?;
+                let cert_csr = cert.certificate_signing_request_string()?;
+
+                create_file_with_user_group(
```

Comment:
I don't think it makes sense having this file owned by the broker user/group as a CSR has nothing to do with the broker. Instead the owner should be the component which is calling the tedge cli (e.g. don't explicitly change the ownership)...CSR are not secrets, so they can be shared.